lib/actions/preload.coffee

```diff
@@ -0,0 +1,206 @@
+###
+Copyright 2016-2017 Resin.io
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+###
+
+_ = require('lodash')
+form = require('resin-cli-form')
+preload = require('resin-preload')
+Promise = require('bluebird')
+resin = require('resin-sdk-preconfigured')
+
+dockerUtils = require('../utils/docker')
+
+LATEST = 'latest'
+
+buildExpandOptions =
+	build:
+		$select: ['id', 'commit_hash', 'push_timestamp']
+		$orderby: 'push_timestamp desc'
+		$filter:
+			status: 'success'
+
+getApplicationsWithSuccessfulBuilds = (resin, deviceType) ->
+	resin.pine.get
+		resource: 'my_application'
+		options:
+			filter:
+				device_type: deviceType
+				build:
+					$any:
+						$alias: 'b'
+						$expr:
+							b:
+								status: 'success'
+			expand:
+				build:
+					$select: [ 'id', 'commit_hash', 'push_timestamp', 'status' ]
+					$orderby: 'push_timestamp desc'
+					# the filter is commented because it causes an api error
+					#$filter:
+					#	status: 'success'
+			select: [ 'id', 'app_name', 'device_type' ]
+			orderby: 'app_name asc'
+	# manual filtering
+	.then (applications) ->
+		applications.forEach (application) ->
+			application.build = application.build.filter (build) ->
+				build.status == 'success'
+		applications
+
+selectApplication = (resin, form, deviceType) ->
+	getApplicationsWithSuccessfulBuilds(resin, deviceType)
+	.then (applications) ->
+		if applications.length == 0
+			expectedError("You have no apps for a '#{deviceType}' device type.")
```

Comment:
You're right, I updated the error message to "You have no apps with successful builds for a '#{deviceType}' device type."